devdoc/object_lifetime_tracker.md

```diff
@@ -127,19 +135,19 @@ static int is_same_object(PDLIST_ENTRY listEntry, const void* object_match_conte
 ### object_lifetime_tracker_register_object
 
 ```c
-MOCKABLE_FUNCTION(, int, object_lifetime_tracker_register_object, OBJECT_LIFETIME_TRACKER_HANDLE, object_lifetime_tracker, const void*, key, void*, object, DESTROY_OBJECT, destroy_object);
+MOCKABLE_FUNCTION(, OBJECT_LIFETIME_TRACKER_REGISTER_OBJECT_RESULT, object_lifetime_tracker_register_object, OBJECT_LIFETIME_TRACKER_HANDLE, object_lifetime_tracker, const void*, key, void*, object, DESTROY_OBJECT, destroy_object, void*, destroy_context);
 ```
-`object_lifetime_tracker_register_object` registers an object belonging to a key with the object lifetime tracker. Once an object is registered with a given `key`, it can be destroyed using the given `destroy_object` destructor when `object_lifetime_tracker_destroy_all_objects_for_key` is called.
+`object_lifetime_tracker_register_object` registers an object belonging to a key with the object lifetime tracker. Once an object is registered with a given `key`, it can be destroyed using the given `destroy_object` destructor when `object_lifetime_tracker_destroy_all_objects_for_key` is called. `object_lifetime_tracker_register_object` does not allow an `object` to be registered with the same `key` multiple times.
```

Comment:
>the same [](http://example.com/codeflow?start=389&length=8)

what about different keys?

Reply:
Yes, that's ok.

That is also desirable. In geo_mux_demux_server we would want to allow the same client to register with multiple destinations.